core/shared/src/main/scala/fs2/text.scala

```diff
@@ -201,4 +204,219 @@ object text {
+      Left(
+        "Malformed padding - final quantum may optionally be padded with one or two padding characters such that the quantum is completed"
+      )
+
+    def decode(state: State, str: String): Either[String, (State, Chunk[Byte])] = {
+      var buffer = state.buffer
+      var mod = state.mod
+      var padding = state.padding
+      var idx, bidx = 0
+      val acc = new Array[Byte]((str.size + 3) / 4 * 3)
+      while (idx < str.length) {
+        str(idx) match {
+          case c if alphabet.ignore(c) => // ignore
+          case c =>
+            val cidx = {
+              if (padding == 0) {
+                if (c == Pad) {
+                  if (mod == 2 || mod == 3) {
+                    padding += 1
+                    0
+                  } else {
+                    return paddingError
+                  }
+                } else {
+                  try alphabet.toIndex(c)
+                  catch {
+                    case _: IllegalArgumentException =>
+                      return Left(s"Invalid base 64 character '$c' at index $idx")
+                  }
+                }
+              } else {
+                if (c == Pad) {
+                  if (padding == 1 && mod == 3) {
+                    padding += 1
+                    0
+                  } else {
+                    return paddingError
+                  }
+                } else {
+                  return Left(
+                    s"Unexpected character '$c' at index $idx after padding character; only '=' and whitespace characters allowed after first padding character"
+                  )
+                }
+              }
+            }
+            mod match {
+              case 0 =>
+                buffer = (cidx & 0x3f)
+                mod += 1
+              case 1 =>
+                buffer = (buffer << 6) | (cidx & 0x3f)
+                mod += 1
+              case 2 =>
+                buffer = (buffer << 6) | (cidx & 0x3f)
+                mod += 1
+              case 3 =>
+                buffer = (buffer << 6) | (cidx & 0x3f)
+                mod = 0
+                val c = buffer & 0x0ff
+                val b = (buffer >> 8) & 0x0ff
+                val a = (buffer >> 16) & 0x0ff
+                acc(bidx) = a.toByte
+                acc(bidx + 1) = b.toByte
+                acc(bidx + 2) = c.toByte
+                bidx += 3
+            }
+        }
+        idx += 1
+      }
+      val out = Chunk.byteVector(ByteVector.view(acc).take((bidx - padding).toLong))
+      val carry = State(buffer, mod, padding)
+      Right((carry, out))
+    }
+
+    def finish(state: State): Either[String, Chunk[Byte]] =
+      if (state.padding != 0 && state.mod != 0) paddingError
+      else
+        state.mod match {
+          case 0 => Right(Chunk.empty)
+          case 1 => Left("Final base 64 quantum had only 1 digit - must have at least 2 digits")
+          case 2 =>
+            Right(Chunk(((state.buffer >> 4) & 0x0ff).toByte))
```

Comment:
Is there a reason this is `0x0ff` vs `0xff`?

Reply:
No, old habit. Fixing.